research/cifar10/evaluate_on_test.py

```diff
@@ -0,0 +1,748 @@
+                )
+                test_results[f"agg_client_{client_number}_pre_best_model_local_avg"] = avg_test_metric
+                test_results[f"agg_client_{client_number}_pre_best_model_local_std"] = std_test_metric
+
+        if eval_last_pre_aggregation_local_models:
+            avg_test_metric, std_test_metric = get_metric_avg_std(pre_last_local_test_metrics)
+            log(
+                INFO,
+                f"""Client {client_number} Pre-aggregation Last Model Average Test
+                Performance on own Data: {avg_test_metric}""",
+            )
+            log(
+                INFO,
+                f"""Client {client_number} Pre-aggregation Last Model St. Dev. Test
+                  Performance on own Data: {std_test_metric}""",
+            )
+            test_results[f"client_{client_number}_pre_last_model_local_avg"] = avg_test_metric
+            test_results[f"client_{client_number}_pre_last_model_local_std"] = std_test_metric
+            if eval_over_aggregated_test_data:
+                avg_test_metric, std_test_metric = get_metric_avg_std(pre_last_local_agg_test_metrics)
+                log(
+                    INFO,
+                    f"""Aggregated Client {client_number} Pre-aggregation Last Model Average Test
+                    Performance on own Data: {avg_test_metric}""",
+                )
+                log(
+                    INFO,
+                    f"""Aggregated Client {client_number} Pre-aggregation Last Model St. Dev. Test
+                    Performance on own Data: {std_test_metric}""",
+                )
+                test_results[f"agg_client_{client_number}_pre_last_model_local_avg"] = avg_test_metric
+                test_results[f"agg_client_{client_number}_pre_last_model_local_std"] = std_test_metric
+
+        if eval_best_post_aggregation_local_models:
+            avg_test_metric, std_test_metric = get_metric_avg_std(post_best_local_test_metrics)
+            log(
+                INFO,
+                f"""Client {client_number} Post-aggregation Best Model Average Test
+                  Performance on own Data: {avg_test_metric}""",
+            )
+            log(
+                INFO,
+                f"""Client {client_number} Post-aggregation Best Model St. Dev. Test
+                  Performance on own Data: {std_test_metric}""",
+            )
+            test_results[f"client_{client_number}_post_best_model_local_avg"] = avg_test_metric
+            test_results[f"client_{client_number}_post_best_model_local_std"] = std_test_metric
+
+            if eval_over_aggregated_test_data:
+                avg_test_metric, std_test_metric = get_metric_avg_std(post_best_local_agg_test_metrics)
+                log(
+                    INFO,
+                    f"""Aggregated Client {client_number} Post-aggregation Best Model Average Test
+                    Performance on own Data: {avg_test_metric}""",
```

Comment:
I might be wrong, but this isn't actually on the clients "own" data, but rather on the entirety of the test dataset? (similar for the logging below)